### PR TITLE
New jtl listener service script for Locust.io

### PR DESCRIPTION
**Pull request by jorgetamayo21**
Added a queue to accumulate results and an array of concurrent greenlets to send it to the listener

### CODE REVIEW

File: scripts/jtl_listener_service.py [hunk 1]
```diff
@@ -34,8 +37,12 @@ def __init__(
         self.listener_url = f"{self.backend_url}:6000"
         # a timestamp format, others could be added...
         self.timestamp_format = timestamp_format
-        # how many records should be held before flushing to disk
+        # how many records should be held before flushing
         self.flush_size = 500
+        self.dequeuers_config = {
+            "max_threads": 50,
```
Comment (ludeknovy) [inline scripts/jtl_listener_service.py:43]: What about putting here like 2 as a default value with a comment to increase it?

File: scripts/jtl_listener_service.py [hunk 2]
```diff
@@ -50,19 +57,46 @@ def __init__(
...
+                "monitor": cpu_usage,
+            })
+        
+    def _dequeue_results(self, name):
+        while True:
+            gevent.sleep(self.dequeuers_config["send_interval"])
+            try:
+                self._log_results(payload=self.queue.get(timeout=10))
+            except:
+                if len(self._background_dequeuers)>1 or self._finished:
```
Comment (ludeknovy) [inline scripts/jtl_listener_service.py:73]: missing spaces?

File: scripts/jtl_listener_service.py [hunk 3]
```diff
@@ -50,19 +57,46 @@ def __init__(
...
+                    logging.info(f"Stoping jtl-reporter {name}")
+                    break
+
+    def _update_dequeuers_threads_quantity(self):
+        while not self._finished:
```
Comment (ludeknovy) [inline scripts/jtl_listener_service.py:78]: I dont have experience with Queue module, but do we really need the while loop to start the dequeuers?

**Review by ludeknovy — CHANGES_REQUESTED** [review]

**Comment by jorgetamayo21** [comment]
These days I have continued developing this script I have added, the final design is:

Sending threads are responsible for processing the payload queue

Receiver threads are responsible for processing the results queue and cpu usage

A thread that manages the number of sending threads, preventing the payload queue from growing

A thread that manages the number of receiving threads, preventing the resuts and cpu usage queue from growing

I will add all this to the script, I will also add comments explaining each thing and I will redo the pull request

**Pull request closed without merging**